crates/holochain/src/conductor/api/api_external/admin_interface.rs

```diff
@@ -83,16 +106,39 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
                 // Install Dnas
                 let tasks = dnas.into_iter().map(|dna_payload| async {
                     let InstallAppDnaPayload {
-                        path,
+                        path: maybe_path,
+                        hash: maybe_hash,
                         properties,
                         membrane_proof,
                         nick,
                     } = dna_payload;
-                    let dna = read_parse_dna(path, properties).await?;
-                    let hash = dna.dna_hash().clone();
-                    let cell_id = CellId::from((hash.clone(), agent_key.clone()));
-                    self.conductor_handle.install_dna(dna).await?;
-                    ConductorApiResult::Ok((InstalledCell::new(cell_id, nick), membrane_proof))
+                    if maybe_path.is_none() && maybe_hash.is_none() {
+                        return Err(ConductorApiError::DnaReadError("Neither path nor hash specified in payload".to_string()))
+                    };
+                    if maybe_path.is_some() && maybe_hash.is_some() {
+                        return Err(ConductorApiError::DnaReadError("Both path and hash specified in payload, pick just one".to_string()))
+                    }
+                    if let Some(path) = maybe_path {
+                        // TODO: this if let will be removed after deprecation period
+                        tracing::info!("specifying dna by path with register side-effect is deprecated, please use RegisterDna and install by hash");
+                        let dna = read_parse_dna(path, properties).await?;
+                        let hash = dna.dna_hash().clone();
+                        let cell_id = CellId::from((hash.clone(), agent_key.clone()));
+                        self.conductor_handle.install_dna(dna).await?;
+                        ConductorApiResult::Ok((InstalledCell::new(cell_id, nick), membrane_proof))
+                    } else if let Some(hash) = maybe_hash {
+                        // confirm that hash has been installed
+                        // TODO: make sure this is only for real DNAs not ones that have
+                        // been installed via properties?
```

Comment:
What does `installed via properties` mean?

Reply:
There's  a question that I didn't test out which is if I register a DNA and override properties how is that stored?  Is it stored twice as two wasms?  Can I get back the properties, or the "progenitor" DNA hash?  This was my question.

Reply:
This got resolved and I added Hash as a DnaSource so you can derive a Dna from an existing one.